lib/generator.js

```diff
@@ -1297,76 +1297,81 @@
+                paddingInputWidth   = paddedInputWidth  - visibleInputWidth,
+                paddingInputHeight  = paddedInputHeight - visibleInputHeight,
+
+                // Designated image size
+                paddedOutputWidthFloat = targetWidth  || (paddedInputWidth * (targetHeight  / paddedInputHeight)),
+                paddedOutputHeightFloat = targetHeight || (paddedInputHeight * (targetWidth / paddedInputWidth)),
+
+                // Effects are not scaled when the transformation is non-uniform
+                paddedFloatRatioDiff = Math.abs(paddedOutputWidthFloat / paddedInputWidth -
+                                                paddedOutputHeightFloat / paddedInputHeight),
+                floatDiffEpislon = 2e-16,
+                effectsScaled       = paddedFloatRatioDiff < floatDiffEpislon,
+
+                paddedOutputWidth   = Math.round(paddedOutputWidthFloat),
+                paddedOutputHeight  = Math.round(paddedOutputHeightFloat),
+
+                paddedOutputScaleX  = paddedOutputWidthFloat  / paddedInputWidth,
+                paddedOutputScaleY  = paddedOutputHeightFloat / paddedInputHeight,
+
+                // How much to scale everything that can be scaled (static + padding, maybe effects)
+                scaleX              = effectsScaled ? paddedOutputScaleX : paddedOutputScaleX +
+                                        (effectsInputWidth  * (paddedOutputScaleX - 1)) /
+                                        (staticInputWidth  + paddingInputWidth),
+                scaleY              = effectsScaled ? paddedOutputScaleY : paddedOutputScaleY +
+                                        (effectsInputHeight * (paddedOutputScaleY - 1)) /
+                                        (staticInputHeight + paddingInputHeight),
+
+                // The expected size of the pixmap returned by Photoshop (does not include padding)
+                visibleOutputWidth  = effectsScaled ? scaleX * visibleInputWidth :
+                                                            scaleX * staticInputWidth + effectsInputWidth,
+                visibleOutputHeight = effectsScaled ? scaleY * visibleInputHeight :
                                                         scaleY * staticInputHeight + effectsInputHeight;
 
-        // The settings for getPixmap
-        return {
-            // For backwards compatibility
-            expectedWidth:  Math.round(visibleOutputWidth),
-            expectedHeight: Math.round(visibleOutputHeight),
-            
-            // For now: absolute scaling only
-            inputRect: {
+            tmpInputRect = {
                 left:   staticInputBounds.left,
                 top:    staticInputBounds.top,
                 right:  staticInputBounds.left + staticInputWidth,
                 bottom: staticInputBounds.top  + staticInputHeight
-            },
-            outputRect: {
+            };
+            tmpOutputRect = {
```

Comment:
update: I see why you have them declared separately, so nevermind that point.  But my "tmp" prefix nit stands.